src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -208,20 +212,29 @@ public static Option blockSize(long value) {
         long blockSize = blockSizeOption == null ? fs.getDefaultBlockSize(p) :
                          blockSizeOption.getValue();
 
-        if (appendIfExistsOption != null && appendIfExistsOption.getValue()
-            && fs.exists(p)) {
-          // Read the file and verify header details
-          try (WALFile.Reader reader =
-                   new WALFile.Reader(conf, WALFile.Reader.file(p), new Reader.OnlyHeaderOption())){
-            if (reader.getVersion() != VERSION[3]) {
-              throw new VersionMismatchException(VERSION[3], reader.getVersion());
+        try {
+          out = fs.create(p, false, bufferSize, replication, blockSize);
+        } catch (FileAlreadyExistsException e) {
+          if (appendIfExistsOption != null && appendIfExistsOption.getValue() && fs.exists(p)) {
```

Comment:
Do we really need to check again if it exists? We already know it existed a moment ago, and even with the `fs.exists(p)` call there is still a race for the `fs.append()` call.